pkg/command/client.go

```diff
@@ -265,6 +266,211 @@ func (c *Client) updatePanelTargets(queryManager *QueryManager, panel *simplejso
+			Overwrite: true,
+			FolderID:  dashboardFull.Meta.Get("folderId").MustInt64(),
+		}); err != nil {
+			return fmt.Errorf("SaveDashboard: %w", err)
+		}
+		c.logd("updated %d panels, skipped %d panels", panelsUpdated, panelsSkipped)
+	}
+
+	return nil
+}
+
+func (c *Client) updatePanelDescription(panel *simplejson.Json, folderTitle, dashboardTitle, rowTitle string, overwrite, dryRun bool) (bool, bool, error) {
+	panelType := panel.Get("type").MustString()
+	panelTitle := panel.Get("title").MustString()
+	currentDesc := panel.Get("description").MustString()
+
+	// Check if we should update this panel
+	shouldUpdate := overwrite || c.isInvalidDescription(currentDesc)
+	if !shouldUpdate {
+		return false, true, nil
+	}
+
+	// Generate new description
+	newDesc := c.generatePanelDescription(folderTitle, dashboardTitle, rowTitle, panelType, panelTitle)
+	if newDesc == currentDesc {
+		return false, true, nil
+	}
+
+	// Update the panel description
+	if !dryRun {
+		panel.Set("description", newDesc)
+	}
+
+	action := "would update"
+	if !dryRun {
+		action = "updated"
+	}
+
+	if rowTitle != "" {
+		c.logd("%s panel description: [%s:%s] in row [%s] -> %s", action, panelType, panelTitle, rowTitle, newDesc)
+	} else {
+		c.logd("%s panel description: [%s:%s] -> %s", action, panelType, panelTitle, newDesc)
+	}
+
+	return true, false, nil
+}
+
+func (c *Client) isInvalidDescription(desc string) bool {
+	if desc == "" {
+		return true
+	}
+
+	// Check if description is just "query=" or similar invalid format
+	queryPaths := c.parseQueryPaths(desc)
+	return len(queryPaths) == 0
+}
+
+func (c *Client) generatePanelDescription(folderTitle, dashboardTitle, rowTitle, panelType, panelTitle string) string {
+	// Sanitize titles
+	sanitizedFolderTitle := c.sanitizeTitle(folderTitle)
+	sanitizedDashboardTitle := c.sanitizeTitle(dashboardTitle)
+	sanitizedPanelTitle := c.sanitizeTitle(panelTitle)
+
+	// Get prefix for panel type
+	prefix := c.getPanelTypePrefix(panelType)
+
+	// Build the path
+	var path string
+	if rowTitle != "" {
+		sanitizedRowTitle := c.sanitizeTitle(rowTitle)
+		path = fmt.Sprintf("%s/%s/%s/%s-%s", sanitizedFolderTitle, sanitizedDashboardTitle, sanitizedRowTitle, prefix, sanitizedPanelTitle)
+	} else {
+		path = fmt.Sprintf("%s/%s/%s-%s", sanitizedFolderTitle, sanitizedDashboardTitle, prefix, sanitizedPanelTitle)
+	}
+
+	return fmt.Sprintf("query=%s", path)
+}
+
+func (c *Client) sanitizeTitle(title string) string {
+	// Convert to lowercase
+	title = strings.ToLower(title)
+
+	// Replace spaces and special characters with hyphens
+	reg := regexp.MustCompile(`[^a-z0-9]+`)
+	title = reg.ReplaceAllString(title, "-")
+
+	// Remove leading/trailing hyphens
+	title = strings.Trim(title, "-")
+
+	// Ensure it's not empty
+	if title == "" {
+		title = "untitled"
+	}
+
+	return title
+}
+
+func (c *Client) getPanelTypePrefix(panelType string) string {
+	// Map panel types to prefixes
+	prefixMap := map[string]string{
+		"table":      "table",
```

Comment:
Why are the keys equal to the map values(minus the timeseries, which is the new version of the graph)?